server/apps/archive/archive_crop.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import superdesk
+from flask import request
+from superdesk import get_resource_service
+from superdesk.resource import Resource
+from .common import item_url
+from superdesk.services import BaseService
+from superdesk.errors import SuperdeskApiError
+from superdesk.media.media_operations import crop_image, process_file_from_stream
+from superdesk.upload import UploadService, url_for_media
+
+
+class ArchiveCropResource(Resource):
+    endpoint_name = 'archive_crop'
+    url = 'archive/<{0}:item_id>/crop/<{0}:crop_name>'.format(item_url)
+    schema = {
+        'CropLeft': {'type': 'integer'},
+        'CropRight': {'type': 'integer'},
+        'CropTop': {'type': 'integer'},
+        'CropBottom': {'type': 'integer'}
+    }
+    datasource = {'source': 'archive'}
+    resource_methods = ['GET', 'POST', 'DELETE']
+    resource_title = endpoint_name
+    privileges = {'POST': 'archive', 'DELETE': 'archive'}
+
+
+class ArchiveCropService(BaseService):
+
+    def create(self, docs, **kwargs):
+        item_id = request.view_args['item_id']
+        crop_name = request.view_args['crop_name'].lower()
+        archive_service = get_resource_service('archive')
```

Comment:
I see your point, thanks for elaborating.

There's no clear-cut answer to that, I personally use my best judgement when such situations arise. The constants and/or enum-ish objects should IMO typically be used in cases similar to the following:

* **application-wide settings**, such as API_URL, DB_STRING, TMP_DIR_NAME, etc.
* arbitrary values that need to be **consistent throughout the application**, e.g. status codes, generic error strings, workflow state names, internal pre-defined limited sets of category/label names (whatever concept a "category" represents), etc.
* values that are reasonably likely to change and appear in multiple places, and we want to make such changes easily at a **single central place**,
* **meaningful aliases** for fixed values, often defined in some standard, e.g. HTTP_OK, KEY_ESC, ERR_SOCKET_TIMEOUT

On the other hand, constants should probably _not_ be (over)used for the following:

* dictionary keys names of "data" objects (i.e. those that represent entities in a system) - the schema of an entity is known, and besides, an object of a custom type should probably be used for that aynway, e.g. Article, SpellChecker, User etc.
* key names for dictionaries holding the request parameters sent by the user - we know what we expect/demand, and since such data must be validated anyway, the validator can perform all the checks for unexpected and/or missing parameters received.

In the end the goal is always to have a more readable and robust code base, but sometimes the opinions vary on where to draw the line between the sensible refactoring and overengineering.